app/src/main/java/com/iic/shopingo/api/APIResult.java

```diff
@@ -12,8 +14,15 @@ public APIResult() {
     errorMessage = null;
   }
 
-  public APIResult(String errorMessage) {
-    success = false;
-    this.errorMessage = errorMessage;
+  public APIResult(String error) {
+    errorMessage = error;
+  }
+
+  public APIResult(ApiSimpleResponse response) {
```

Comment:
The classes extending `ApiResult` contain the models used in the app, whereas all the `XResponse` classes are used for parsing the response from the server. (The converters then create the appropriate classes from them)